snuba/utils/streams/backends/kafka.py

```diff
@@ -615,26 +618,120 @@ def closed(self) -> bool:
 
 DEFAULT_QUEUED_MAX_MESSAGE_KBYTES = 50000
 DEFAULT_QUEUED_MIN_MESSAGES = 10000
+DEFAULT_PARTITIONER = "consistent"
+DEFAULT_MAX_MESSAGE_BYTES = 50000000  # 50MB, default is 1MB
+SUPPORTED_KAFKA_CONFIGURATION = (
+    # Check https://github.com/edenhill/librdkafka/blob/master/CONFIGURATION.md
+    # for the full list of available options
+    "bootstrap.servers",
+    "sasl.mechanism",
+    "sasl.username",
+    "sasl.password",
+    "security.protocol",
+)
+
+
+def get_default_kafka_configuration(
+    storage_key: Optional[StorageKey] = None,
+    bootstrap_servers: Optional[Sequence[str]] = None,
+    override_params: Optional[Mapping[str, Any]] = None,
+) -> KafkaBrokerConfig:
+    default_bootstrap_servers = None
+    if storage_key is not None:
+        storage_name = storage_key.value
+        if storage_name in settings.DEFAULT_STORAGE_BROKERS:
+            # this is now deprecated
+            logger.warning(
+                "DEPRECATED: DEFAULT_STORAGE_BROKERS is defined. Please use STORAGE_BROKER_CONFIG instead"
+            )
+            default_config = {}
+            default_bootstrap_servers = ",".join(
+                settings.DEFAULT_STORAGE_BROKERS[storage_name]
+            )
+        elif settings.DEFAULT_BROKERS:
+            logger.warning(
+                "DEPRECATED: DEFAULT_BROKERS is defined. Please use BROKER_CONFIG instead"
+            )
+            default_config = {}
+            default_bootstrap_servers = ",".join(settings.DEFAULT_BROKERS)
+        else:
+            default_config = settings.STORAGE_BROKER_CONFIG.get(
+                storage_name, settings.BROKER_CONFIG
+            )
+    else:
+        if settings.DEFAULT_BROKERS:
+            logger.warning(
+                "DEPRECATED: DEFAULT_BROKERS is defined. Please use BROKER_CONFIG instead"
+            )
+            default_config = {}
+            default_bootstrap_servers = ",".join(settings.DEFAULT_BROKERS)
+        else:
+            default_config = settings.BROKER_CONFIG
+    broker_config = default_config.copy()
+    bootstrap_servers = bootstrap_servers or default_bootstrap_servers
+    if bootstrap_servers:
+        broker_config["bootstrap.servers"] = bootstrap_servers
+    broker_config = {k: v for k, v in broker_config.items() if v is not None}
+    for configuration_key in broker_config:
+        if configuration_key not in SUPPORTED_KAFKA_CONFIGURATION:
+            raise RuntimeError(
```

Comment:
I think this is a good candidate for ValueError: https://docs.python.org/3/library/exceptions.html#ValueError